### PR TITLE
Docs

### PR DESCRIPTION
**Pull request by garyb**
Still working on this, but wanted to get some feedback. How does it look? Is it targeting the right level? Would it be better if we used concrete values in the examples, rather than showing them "algebraically"? Is it too much to go into use of applicative style, etc.

Also, `psc-docs` sorting the output alphabetically is quite unhelpful when doing things this way, as I was trying to build on info from `Functor` with `Apply`, etc. but obviously that doesn't work so well in the output.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,50 +4,228 @@
...
+
     instance extendMaybe :: Extend Maybe
 
+#### `functorMaybe`
+
+The `Functor` instance for `Maybe` allows functions to transform the
+contents of a `Just` with the `<$>` operator:
+
+``` purescript
+Just x <$> f == Just (f x)
```
Comment (joneshf) [inline README.md]: These are reversed.

Comment (garyb) [inline README.md]: Hah, how did I get that wrong.

Comment (joneshf) [inline README.md]: `(<#>)` on the brain?

**Comment by paf31** [comment]
I really like the new docs :)

This looks good.

One passing comment - I wonder if it is better to explain `Apply` in terms of things like `liftA2`, or in terms of something like `zip`. To me, `<*>` is either "a way to lift pure functions of `N >= 2` arguments to take arguments of type `Maybe a`, or, possibly even simpler, `zip` for lists of length 0 or 1.

**Comment by joneshf** [comment]
Maybe that explanation is best left for the actual typeclass. It explains things more generally, whereas I would think we'd want to be more specific for an actual data type.

**Comment by garyb** [comment]
Does the case of "pure functions to take arguments of type `Maybe a`" not come across in the latter part of the `Apply` description then? Although that mentions using it with `<$>`, that was what I was trying to convey.

**Comment by paf31** [comment]
I think it does. I would probably just change "... to a function that does not usually expect them". We're not passing `Maybe a` where we expect an `a`, but turning a function of type `a -> b -> ...` into a function `Maybe a -> Maybe b -> ...`, right?

**Comment by garyb** [comment]
Ah right, I see what you mean. Makes sense :+1:

**Comment by paf31** [comment]
Not a complaint at all, by the way. I think this is great.

**Comment by garyb** [comment]
Alright, done now I think. I carried on commenting the instances although that may be a little redunant, as well, I'd already done most of them. ;)

**Comment by paf31** [comment]
Looks great! :ship:

**Pull request merged**